src/posts/2021-09-27-Using-focus-style-for-accessibility

```diff
@@ -0,0 +1,133 @@
+    source:  Caitlin Geier
+    year: 2016
+  - title: Never remove CSS outlines
+    url: https://www.a11yproject.com/posts/2013-01-25-never-remove-css-outlines/
+    source: Guilherme Simões
+    year: 2013
+tags:
+  - How-to
+---
+
+While accessing any digital interface, a focus indicator is important it highlights the active intractable component in a webpage. 
+
+## What is a focus style?
+
+A focus is a pseudo-class used to indicate or highlights an intractable component in a webpage.
+
+## Why is it important?
+
+A focus style is necessary because some people use a mouse or a touch device to access a website some rely on using a keyboard to navigate on a website. Having a focus indicator will improve the usability of your website and makes it accessible, It also tells us the interactive element we are focused on.
+
+## How to apply focus style
+
+Preceding code is an example of how we can apply focus style to our elements:
+
+### Applying focus style to a link
+
+```CSS
+a:focus {
+  outline: 5px solid red;
+  }
+```
+
+The following code example is a link that has a custom style for it focus state:
+
+### Code explanation
+
+In the following code example, we added focus to our link and gave it an outline of solid red with a value of 5px. Please note the focused style is applied only to the link tag itself.
+
+The preceding code is another example for applying focus to a form input:
+
+### Applying focus style to a form input
+
+```HTML
+<form>
+<label for="email"> Email :</label>
+<input class="email-input" value="Click me to see my focus state">
+</form>
+```
+
+```CSS
+.email-input:focus {
+ background-color: gray;
+ color: black;
+}
+```
+
+You can create your custom outline style aside from the blue outline default style in focus. Always ensure the color contrast for your focus indicator is sufficient especially important for people with low vision. This will also benefit those using their screen outside in a very bright space.
+
+## Focus-visible
+
+The `:focus-visible` is a great idea for controlling how users interact with your elements it is used to provide a focus-indicator selectively. 
+
+Let’s look at a quick code example:
+
+```HTML
+<form class="form-container">
+<label for="email">Email:</label>
+<input class="email-input" value="Click me to see my focus state">
+<label for="username">Username:</label>
+<input class="user-input" value="Click me to see my focus state">
+<button type="submit">Submit</button>
+</form>
+```
+
+```CSS
+:focus {
+ outline: 2px solid orange
+}
+```
+
+In the following code example, when we focus on each of the input and buttons it gives us a red outline but focus style can also affect the user experience of mouse users.
+
+It is wrong to remove the focus indicator as it makes your website inaccessible for people navigating with the keyboard the best way is to apply a different focus style for mouse/pointer users.
+
+Code example:
+
+```CSS
+:focus:not(:focus-visible) {
+ outline:none
+}
+```
```

Comment:
I would like to revise this code example to use something other than `:focus:not(:focus-visible)`. I consider this approach an antipattern, as it adjusts focus behavior to something unexpected. This has created confusion for people with lower tech literacy who rely on focus indicators, as well as creates a separate class of experience for non-mouse users, which isn't good.